src/engine/config.rs

```diff
@@ -1266,6 +1266,10 @@ impl Type {
         self.props.iter()
     }
 
+    pub fn mut_props(&mut self) -> &mut Vec<Property> {
```

Comment:
Exposed props as mutable to enable modifications during before_engine_build hooks.